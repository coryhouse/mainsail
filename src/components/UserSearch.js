import React from 'react';
import PropTypes from 'prop-types';

class UserSearch extends React.Component {
  render() {
    const {onSearch, onClear, onChange, firstName, lastName} = this.props;
    return (
      <form>
        <label>First Name</label>
        <input type="text" className="uppercase" name="firstName" onChange={onChange} value={firstName} />

        <label>Last Name</label>
        <input type="text" name="lastName" onChange={onChange} value={lastName} />

        <input type="submit" onClick={onSearch} value="Search" />
        <button type="button" onClick={onClear}>Clear</button>
      </form>
    )
  }
}

UserSearch.propTypes = {
  onChange: PropTypes.func.isRequired,
  onSearch: PropTypes.func.isRequired,
  firstName: PropTypes.string.isRequired,
  lastName: PropTypes.string.isRequired
};

export default UserSearch;