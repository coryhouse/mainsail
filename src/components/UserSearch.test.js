import React from "react";
import UserSearch from "./UserSearch";
import renderer from "react-test-renderer";

it("should render 2 empty search fields", () => {
  const tree = renderer
    .create(<UserSearch firstName='Mark' lastName='Longman' onSearch={jest.fn} onChange={jest.fn} onClear={jest.fn} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});