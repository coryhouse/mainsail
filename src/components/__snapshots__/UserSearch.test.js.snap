// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 2 empty search fields 1`] = `
<form>
  <label>
    First Name
  </label>
  <input
    className="uppercase"
    name="firstName"
    onChange={[Function]}
    type="text"
    value="Mark"
  />
  <label>
    Last Name
  </label>
  <input
    name="lastName"
    onChange={[Function]}
    type="text"
    value="Longman"
  />
  <input
    onClick={[Function]}
    type="submit"
    value="Search"
  />
  <button
    onClick={[Function]}
    type="button"
  >
    Clear
  </button>
</form>
`;
